src/combinator.rs

```diff
@@ -1007,6 +1007,46 @@ pub fn try<P>(p: P) -> Try<P>
 }
 
 #[derive(Clone)]
+pub struct LookAhead<P>(P);
+
+impl <I, O, P> Parser for LookAhead<P>
+    where I: Stream
+        , P: Parser<Input=I, Output=O> {
+
+    type Input = I;
+    type Output = O;
+
+    fn parse_lazy(&mut self, input: State<I>) -> ParseResult<O, I> {
+        self.0.parse_lazy(input.clone()).map(|(o, _input)| (o, Consumed::Empty(input)))
+    }
+
+    fn add_error(&mut self, errors: &mut ParseError<Self::Input>) {
+        self.0.add_error(errors);
+    }
+}
+
+///look_ahead acts as `p`, however doesn't consume input
```

Comment:
This might be better worded as:

`look_ahead` acts as `p` but doesn't consume input on success.

http://hackage.haskell.org/package/parsec-3.1.9/docs/Text-Parsec-Prim.html#v:lookAhead